daemon/src/identify/dialer.rs

```diff
@@ -0,0 +1,176 @@
+                return;
+            }
+        };
+
+        let wire_version = identity_info.wire_version.clone();
+        let daemon_version = identity_info.daemon_version.clone();
+        let environment = identity_info.environment;
+
+        NUM_LIBP2P_CONNECTIONS_GAUGE
+            .with(&HashMap::from([
+                (WIRE_VERSION_LABEL, wire_version.as_str()),
+                (DAEMON_VERSION_LABEL, daemon_version.as_str()),
+                (ENVIRONMENT_LABEL, environment.to_string().as_str()),
+            ]))
+            .inc();
+
+        tracing::info!(%peer_id, %daemon_version, %environment, %wire_version, "New identify message received");
+        self.identities.insert(peer_id, identity_info);
+    }
+}
+
+#[xtra_productivity(message_impl = false)]
+impl Actor {
+    async fn handle_connections_established(
+        &mut self,
+        msg: endpoint::ConnectionEstablished,
+        ctx: &mut Context<Self>,
+    ) {
+        let peer_id = msg.peer;
+        let endpoint = self.endpoint.clone();
+        let this = ctx.address().expect("we are alive");
+
+        let request_peer_identity_fut = {
+            let this = this.clone();
+            async move {
+                let stream = endpoint
+                    .send(OpenSubstream::single_protocol(peer_id, PROTOCOL))
+                    .await??;
+
+                let peer_identity = protocol::recv(stream).await?;
+
+                this.send(PeerIdentityReceived {
+                    peer_id,
+                    peer_identity,
+                })
+                .await?;
+
+                anyhow::Ok(())
+            }
+        };
+
+        let err_handler = move |e| async move {
+            tracing::debug!(%peer_id, "Identity protocol failed upon request: {e:#}")
+        };
+
+        spawn_fallible(&this, request_peer_identity_fut, err_handler);
+    }
+
+    async fn handle_connections_dropped(&mut self, msg: endpoint::ConnectionDropped) {
+        let peer_id = msg.peer;
+        tracing::trace!(%peer_id, "Remove peer-info because connection dropped");
+        if let Some(identity_info) = self.identities.remove(&peer_id) {
+            NUM_LIBP2P_CONNECTIONS_GAUGE
+                .with(&HashMap::from([
+                    (WIRE_VERSION_LABEL, identity_info.wire_version.as_str()),
+                    (DAEMON_VERSION_LABEL, identity_info.daemon_version.as_str()),
+                    (
+                        ENVIRONMENT_LABEL,
+                        identity_info.environment.to_string().as_str(),
+                    ),
+                ]))
+                .dec();
+        }
+    }
+}
+
+const WIRE_VERSION_LABEL: &str = "wire_version";
+const DAEMON_VERSION_LABEL: &str = "daemon_version";
+const ENVIRONMENT_LABEL: &str = "environment";
+
+static NUM_LIBP2P_CONNECTIONS_GAUGE: conquer_once::Lazy<prometheus::IntGaugeVec> =
+    conquer_once::Lazy::new(|| {
+        prometheus::register_int_gauge_vec!(
+            "libp2p_connections_total",
```

Comment:
❓ Is this the only place where we are tracking the number of connections via libp2p?

Reply:
Maybe the name is misleading, I tried to mimic the previous functionality, so we are able to track takers and versions. This is actually to some extent independent to the connection as such.